### ISSUE TITLE
trouble with sharp/jimp

### ISSUE THREAD

**Issue by Shawak** [oven-sh/bun#13846]
### What version of Bun is running?

1.1.27

### What platform is your computer?

debian/alpine docker container

### What steps can reproduce the bug?

Upgrade to 1.1.27 and use the jimp package, e.g.:

```typescript
import Jimp from 'jimp';
import sharp from 'sharp';

  const buffer = await (async () => {
      const response = await fetch(imageLink);
      const arrayBuffer = await response.arrayBuffer();
      return await sharp(arrayBuffer).toFormat('png').toBuffer();
  })();

  const image = await Jimp.read(buffer);
  const imageBuffer = await image
      .invert()
      .greyscale()
      .getBufferAsync(Jimp.MIME_PNG);
```

### What do you see instead?

```
Date: Tue Sep 10 2024 07:03:51 GMT+0000 (Coordinated Universal Time)
106 |   }
107 | 
108 |   assert(this._handle, "zlib binding closed");
109 |   let res;
110 |   do {
111 |     res = this._handle.writeSync(
                             ^
TypeError: this._handle.writeSync is not a function. (In 'this._handle.writeSync(flushFlag, chunk, inOff, availInBefore, this._buffer, this._offset, availOutBefore)', 'this._handle.writeSync' is undefined)
      at /home/bun/app/node_modules/pngjs/lib/sync-inflate.js:111:24
      at /home/bun/app/node_modules/pngjs/lib/parser-sync.js:85:20
      at /home/bun/app/node_modules/@jimp/core/dist/utils/image-bitmap.js:157:47
```

**Comment by nektro** [comment]
changing first line to `import { Jimp } from "jimp";` fails with

```
5 |   const response = await fetch(imageLink);
                                            ^
ReferenceError: Can't find variable: imageLink
      at /Users/meghandenny/src/test/index.js:5:41
```

**Comment by tobias-theobald** [comment]
I have created a minimal reproduction on [replit](https://replit.com/@replitdivisible/Bun-Issue-13846-Replication-Jimp-cannot-read-pngs?v=1). This does work with node/tsx.

Note how the crash actually happens in [pngjs](https://github.com/pngjs/pngjs/blob/main/lib/sync-inflate.js#L111) but it's inheriting the `Inflate` class from zlib (a node-native module) and doing something weird with underscore-prefixed methods, a notation which should be reserved for internals. [zlib should be supported by bun](https://bun.sh/docs/runtime/nodejs-apis#node-zlib), that is how I understand the claim that it's not optimized though.

**Comment by nektro** [comment]
excellent, thanks. can confirm this is now fixed by the latest canary and will be available again in 1.1.30
